TelepresenceBot/server-unit/core/server.js

```diff
@@ -72,25 +71,29 @@ io.sockets.on('connection', function (client) {
             testClient.emit('direction', direction);
         }
     });
+});
 
-    function leaveAllRooms(client) {
-        var rooms = Object.keys(io.sockets.adapter.sids[client.id]);
-        for(var i = 0; i < rooms.length; i++) {
-            client.leave(rooms[i]);
+function disconnectRoom(name) {
+    var room = io.sockets.adapter.rooms[name];
+
+    if(room != undefined) {
+        var clients = io.sockets.adapter.rooms[name].sockets;
+
+        for(var client in clients) {
+            var connectedClient = io.sockets.connected[client];
+            connectedClient.disconnect();
         }
     }
+}
 
-    function disconnectRoom(name) {
-        var room = io.sockets.adapter.rooms[name];
+function asRoomsWithSocketIds() {
```

Comment:
The tests now output an object with property keys that match a room name and values that match the sockets in the room.